content/recipes/prisma.md

```diff
@@ -248,24 +248,18 @@ When setting up your NestJS application, you'll want to abstract away the Prisma
 Inside the `src` directory, create a new file called `prisma.service.ts` and add the following code to it:
 
 ```typescript
-import { INestApplication, Injectable, OnModuleInit } from '@nestjs/common';
+import { Injectable, OnModuleInit } from '@nestjs/common';
 import { PrismaClient } from '@prisma/client';
 
 @Injectable()
 export class PrismaService extends PrismaClient implements OnModuleInit {
   async onModuleInit() {
     await this.$connect();
   }
-
-  async enableShutdownHooks(app: INestApplication) {
-    this.$on('beforeExit', async () => {
-      await app.close();
-    });
-  }
 }
 ```
 
-> info **Note** The `onModuleInit` is optional — if you leave it out, Prisma will connect lazily on its first call to the database. We don't bother with `onModuleDestroy`, since Prisma has its own shutdown hooks where it will destroy the connection. For more info on `enableShutdownHooks`, please see [Issues with `enableShutdownHooks`](recipes/prisma#issues-with-enableshutdownhooks)
```

Comment:
> We don't bother with `onModuleDestroy`, since Prisma has its own shutdown hooks where it will destroy the connection.

This was also a little inaccurate, at least since Prisma 3.

Prisma's own shutdown hooks in Prisma 2 (or in modern Prisma when using the binary engine) existed to kill the query engine process, but they didn't explicitly close the db connection. I'm not sure if the engine did that by catching the signal itself, but I can't seem to find that in the engines codebase.

In Prisma 3+, with library engine (a Node.js native addon), the only reason the internal shutdown hooks still existed was to maintain backward compatibility and keep the `beforeExit` event, but they haven't been doing anything beyond that.

If it's desired to close the connections gracefully rather than dropping them, it's still necessary to add an `onModuleDestroy` method with explicit `$disconnect`.